### PR TITLE
ramips/mt7621: enable linux 5.10, bump to 5.10.16

### PR DESCRIPTION
**Pull request by lipnitsk**
Enable Linux 5.10 on ramips target, resolve various build conflicts. Make a bunch of modules work with it. Runs well on my Ubiquity Edgerouter ER-X. May be some duplication with https://github.com/openwrt/openwrt/pull/3693/, but this is fresher :)

### CODE REVIEW

**Comment by timocapa** [comment]
ramips/mt7621/xiaomi mi r3g

"Strip unnecessary exports from the kernel image" doesn't compile (no log right now because busy), and i get major packet loss. downgraded to 5.4.X and it seems fine again :(

**Comment by adschm** [comment]
This is a duplicate of https://github.com/openwrt/openwrt/pull/3693

**Comment by blocktrron** [comment]
@lipnitsk I've picked ebba9fbe5d6b96e52475a924129b339aa97bdcb4 to my staging tree and will merge it after a test build on K5.10 and K5.4. Thanks!

**Comment by lipnitsk** [comment]
@blocktrron thanks, I did put quite a bit of work into these, so I'm glad that the module changes interested you :) FYI, I updated it just a bit for that commit to be standalone, see 6a15c39235489590a388f155e5fb38b5772184ee - you may want to pick that instead.

Also, let me know if you want `CONFIG_LINUX_5_4` clauses replaced with `@lt5.10` or similar.

**Comment by lipnitsk** [comment]
> ramips/mt7621/xiaomi mi r3g
> 
> "Strip unnecessary exports from the kernel image" doesn't compile (no log right now because busy), and i get major packet loss. downgraded to 5.4.X and it seems fine again :(

Is this with my branch? Please post the log when you get a chance.

**Comment by timocapa** [comment]
> > ramips/mt7621/xiaomi mi r3g
> > "Strip unnecessary exports from the kernel image" doesn't compile (no log right now because busy), and i get major packet loss. downgraded to 5.4.X and it seems fine again :(
> 
> Is this with my branch? Please post the log when you get a chance.

Device/kernel log looked fine, i still had major packet loss though, i'll look into it later again when i have time and everyone is asleep.

Build log with stripping:

```openwrt-r3g on  HEAD [!] took 9s 848ms 
➜ make -j1 V=sc
make[2]: Entering directory '/home/timo/Git/openwrt-r3g/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/home/timo/Git/openwrt-r3g/scripts/config'
make[1]: Entering directory '/home/timo/Git/openwrt-r3g'
make[2]: Entering directory '/home/timo/Git/openwrt-r3g'
make[3]: Entering directory '/home/timo/Git/openwrt-r3g/target/linux'
make[4]: Entering directory '/home/timo/Git/openwrt-r3g/target/linux/ramips'
rm -f /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/localversion
/home/timo/Git/openwrt-r3g/scripts/kconfig.pl  + /home/timo/Git/openwrt-r3g/target/linux/generic/config-5.10 /home/timo/Git/openwrt-r3g/target/linux/ramips/mt7621/config-5.10 > /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target
awk '/^(#[[:space:]]+)?CONFIG_KERNEL/{sub("CONFIG_KERNEL_","CONFIG_");print}' /home/timo/Git/openwrt-r3g/.config >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target
echo "# CONFIG_KALLSYMS_EXTRA_PASS is not set" >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target
echo "# CONFIG_KALLSYMS_ALL is not set" >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target
echo "CONFIG_KALLSYMS_UNCOMPRESSED=y" >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target
/home/timo/Git/openwrt-r3g/scripts/package-metadata.pl kconfig /home/timo/Git/openwrt-r3g/tmp/.packageinfo /home/timo/Git/openwrt-r3g/.config 5.10 > /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.override
/home/timo/Git/openwrt-r3g/scripts/kconfig.pl 'm+' '+' /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.target /dev/null /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.override > /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set
mv /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.old
grep -v INITRAMFS /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.old > /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set
echo 'CONFIG_INITRAMFS_SOURCE=""' >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set
echo '# CONFIG_INITRAMFS_FORCE is not set' >> /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set
rm -rf /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/modules
cmp -s /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.prev || { cp /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config; cp /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.prev; }
export MAKEFLAGS= ; [ -d /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/user_headers ] || make -C /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16 KCFLAGS="-fmacro-prefix-map=/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl=target-mipsel_24kc_musl" HOSTCFLAGS="-Ofast -I/home/timo/Git/openwrt-r3g/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="mipsel-openwrt-linux-musl-" ARCH="mips" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Feb 17 10:46:54 2021" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/home/timo/Git/openwrt-r3g/staging_dir/host/lib" KBUILD_HOSTLDLIBS="-L/home/timo/Git/openwrt-r3g/staging_dir/host/lib" CONFIG_SHELL="bash" V=1  cmd_syscalls=  KERNELRELEASE=5.10.16 EXTRA_LDSFLAGS="-I/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621 -include symtab.h" CC="mipsel-openwrt-linux-musl-gcc" INSTALL_HDR_PATH=/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/user_headers headers_install
grep '=[ym]' /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.config.set | LC_ALL=C sort | mkhash md5 > /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.vermagic
touch /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.configured
rm -f /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/vmlinux /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/System.map
make -C /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16 KCFLAGS="-fmacro-prefix-map=/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl=target-mipsel_24kc_musl" HOSTCFLAGS="-Ofast -I/home/timo/Git/openwrt-r3g/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="mipsel-openwrt-linux-musl-" ARCH="mips" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Feb 17 10:46:54 2021" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/home/timo/Git/openwrt-r3g/staging_dir/host/lib" KBUILD_HOSTLDLIBS="-L/home/timo/Git/openwrt-r3g/staging_dir/host/lib" CONFIG_SHELL="bash" V=1  cmd_syscalls=  KERNELRELEASE=5.10.16 EXTRA_LDSFLAGS="-I/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621 -include symtab.h" CC="mipsel-openwrt-linux-musl-gcc" vmlinux vmlinuz modules
make[5]: Entering directory '/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16'
make -f ./scripts/Makefile.build obj=arch/mips/kernel/syscalls all
make -f ./scripts/Makefile.build obj=scripts/basic
rm -f .tmp_quiet_recordmcount
make -f ./scripts/Makefile.build obj=arch/mips/tools elf-entry
make -f ./scripts/Makefile.build obj=arch/mips/boot/tools relocs
make -f ./scripts/Makefile.build obj=scripts/dtc
make -f ./scripts/Makefile.build obj=scripts
set -e; mkdir -p include/config/; trap "rm -f include/config/.kernel.release.tmp" EXIT; { echo "5.10.16$(bash ./scripts/setlocalversion .)"; } > include/config/.kernel.release.tmp; if [ ! -r include/config/kernel.release ] || ! cmp -s include/config/kernel.release include/config/.kernel.release.tmp; then : '  UPD     include/config/kernel.release'; mv -f include/config/.kernel.release.tmp include/config/kernel.release; fi
make -f ./scripts/Makefile.asm-generic obj=arch/mips/include/generated/uapi/asm \
generic=include/uapi/asm-generic
make -f ./scripts/Makefile.asm-generic obj=arch/mips/include/generated/asm \
generic=include/asm-generic
set -e; mkdir -p include/generated/uapi/linux/; trap "rm -f include/generated/uapi/linux/.version.h.tmp" EXIT; { 	echo \#define LINUX_VERSION_CODE 330256; echo '#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))'; } > include/generated/uapi/linux/.version.h.tmp; if [ ! -r include/generated/uapi/linux/version.h ] || ! cmp -s include/generated/uapi/linux/version.h include/generated/uapi/linux/.version.h.tmp; then : '  UPD     include/generated/uapi/linux/version.h'; mv -f include/generated/uapi/linux/.version.h.tmp include/generated/uapi/linux/version.h; fi
rm -f include/linux/version.h
set -e; mkdir -p include/generated/; trap "rm -f include/generated/.utsrelease.h.tmp" EXIT; { 	if [ `echo -n "5.10.16" | wc -c ` -gt 64 ]; then echo '"5.10.16" exceeds 64 characters' >&2; exit 1; fi; echo \#define UTS_RELEASE \"5.10.16\"; } > include/generated/.utsrelease.h.tmp; if [ ! -r include/generated/utsrelease.h ] || ! cmp -s include/generated/utsrelease.h include/generated/.utsrelease.h.tmp; then : '  UPD     include/generated/utsrelease.h'; mv -f include/generated/.utsrelease.h.tmp include/generated/utsrelease.h; fi
make -f ./scripts/Makefile.build obj=scripts/mod
set -e; mkdir -p scripts/mod/; trap "rm -f scripts/mod/.devicetable-offsets.h.tmp" EXIT; { 	 echo "#ifndef __DEVICETABLE_OFFSETS_H__"; echo "#define __DEVICETABLE_OFFSETS_H__"; echo "/*"; echo " * DO NOT MODIFY."; echo " *"; echo " * This file was generated by Kbuild"; echo " */"; echo ""; sed -ne 	's:^[[:space:]]*\.ascii[[:space:]]*"\(.*\)".*:\1:; /^->/{s:->#\(.*\):/* \1 */:; s:^->\([^ ]*\) [\$#]*\([^ ]*\) \(.*\):#define \1 \2 /* \3 */:; s:->::; p;}' < scripts/mod/devicetable-offsets.s; echo ""; echo "#endif"; } > scripts/mod/.devicetable-offsets.h.tmp; if [ ! -r scripts/mod/devicetable-offsets.h ] || ! cmp -s scripts/mod/devicetable-offsets.h scripts/mod/.devicetable-offsets.h.tmp; then : '  UPD     scripts/mod/devicetable-offsets.h'; mv -f scripts/mod/.devicetable-offsets.h.tmp scripts/mod/devicetable-offsets.h; fi
make -f ./scripts/Makefile.build obj=.
set -e; mkdir -p include/generated/; trap "rm -f include/generated/.bounds.h.tmp" EXIT; { 	 echo "#ifndef __LINUX_BOUNDS_H__"; echo "#define __LINUX_BOUNDS_H__"; echo "/*"; echo " * DO NOT MODIFY."; echo " *"; echo " * This file was generated by Kbuild"; echo " */"; echo ""; sed -ne 	's:^[[:space:]]*\.ascii[[:space:]]*"\(.*\)".*:\1:; /^->/{s:->#\(.*\):/* \1 */:; s:^->\([^ ]*\) [\$#]*\([^ ]*\) \(.*\):#define \1 \2 /* \3 */:; s:->::; p;}' < kernel/bounds.s; echo ""; echo "#endif"; } > include/generated/.bounds.h.tmp; if [ ! -r include/generated/bounds.h ] || ! cmp -s include/generated/bounds.h include/generated/.bounds.h.tmp; then : '  UPD     include/generated/bounds.h'; mv -f include/generated/.bounds.h.tmp include/generated/bounds.h; fi
set -e; mkdir -p include/generated/; trap "rm -f include/generated/.timeconst.h.tmp" EXIT; { echo 250 | bc -q kernel/time/timeconst.bc; } > include/generated/.timeconst.h.tmp; if [ ! -r include/generated/timeconst.h ] || ! cmp -s include/generated/timeconst.h include/generated/.timeconst.h.tmp; then : '  UPD     include/generated/timeconst.h'; mv -f include/generated/.timeconst.h.tmp include/generated/timeconst.h; fi
set -e; mkdir -p include/generated/; trap "rm -f include/generated/.asm-offsets.h.tmp" EXIT; { 	 echo "#ifndef __ASM_OFFSETS_H__"; echo "#define __ASM_OFFSETS_H__"; echo "/*"; echo " * DO NOT MODIFY."; echo " *"; echo " * This file was generated by Kbuild"; echo " */"; echo ""; sed -ne 	's:^[[:space:]]*\.ascii[[:space:]]*"\(.*\)".*:\1:; /^->/{s:->#\(.*\):/* \1 */:; s:^->\([^ ]*\) [\$#]*\([^ ]*\) \(.*\):#define \1 \2 /* \3 */:; s:->::; p;}' < arch/mips/kernel/asm-offsets.s; echo ""; echo "#endif"; } > include/generated/.asm-offsets.h.tmp; if [ ! -r include/generated/asm-offsets.h ] || ! cmp -s include/generated/asm-offsets.h include/generated/.asm-offsets.h.tmp; then : '  UPD     include/generated/asm-offsets.h'; mv -f include/generated/.asm-offsets.h.tmp include/generated/asm-offsets.h; fi
  bash scripts/atomic/check-atomics.sh
make -f ./scripts/Makefile.build obj=init \
single-build= \
need-builtin=1 need-modorder=1
bash ./scripts/mkcompile_h include/generated/compile.h	\
"mips" "y" ""	\
"" "mipsel-openwrt-linux-musl-gcc (OpenWrt GCC 10.2.0 r15829+15-482c9ff289) 10.2.0" "mipsel-openwrt-linux-musl-ld"
make -f ./scripts/Makefile.build obj=usr \
single-build= \
need-builtin=1 need-modorder=1
make -f ./scripts/Makefile.build obj=arch/mips \
single-build= \
need-builtin=1 need-modorder=1
make -f ./scripts/Makefile.build obj=arch/mips/kernel \
 \
need-builtin=1 \
need-modorder=1
  mipsel-openwrt-linux-musl-gcc -E -I/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621 -include symtab.h -Wp,-MMD,arch/mips/kernel/.vmlinux.lds.d -nostdinc -isystem /home/timo/Git/openwrt-r3g/staging_dir/toolchain-mipsel_24kc_gcc-10.2.0_musl/lib/gcc/mipsel-openwrt-linux-musl/10.2.0/include -I./arch/mips/include -I./arch/mips/include/generated  -I./include -I./arch/mips/include/uapi -I./arch/mips/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -DVMLINUX_LOAD_ADDRESS=0xffffffff80001000 -DLINKER_LOAD_ADDRESS=0x80001000 -DDATAOFFSET=0 -fmacro-prefix-map=./=  -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -fno-PIE -Werror=implicit-function-declaration -Werror=implicit-int -Werror=return-type -Wno-format-security -std=gnu89  -mno-check-zero-division -mabi=32 -G 0 -mno-abicalls -fno-pic -pipe -mno-branch-likely -msoft-float -DGAS_HAS_SET_HARDFLOAT -Wa,-msoft-float -ffreestanding -EL -fno-stack-check -march=mips32r2 -mtune=34kc -Wa,--trap -DTOOLCHAIN_SUPPORTS_VIRT -DTOOLCHAIN_SUPPORTS_XPA -DTOOLCHAIN_SUPPORTS_CRC -DTOOLCHAIN_SUPPORTS_DSP -DTOOLCHAIN_SUPPORTS_GINV -I./arch/mips/include/asm/mach-ralink -I./arch/mips/include/asm/mach-ralink/mt7621 -I./arch/mips/include/asm/mach-generic -fno-asynchronous-unwind-tables -fno-delete-null-pointer-checks -Wno-frame-address -Wno-format-truncation -Wno-format-overflow -Wno-address-of-packed-member -O2 -fno-caller-saves -fno-allow-store-data-races -Wframe-larger-than=1024 -fstack-protector -Wno-unused-but-set-variable -Wimplicit-fallthrough -Wno-unused-const-variable -fomit-frame-pointer -Wdeclaration-after-statement -Wvla -Wno-pointer-sign -Wno-stringop-truncation -Wno-zero-length-bounds -Wno-array-bounds -Wno-stringop-overflow -Wno-restrict -Wno-maybe-uninitialized -fno-strict-overflow -fno-stack-check -fconserve-stack -Werror=date-time -Werror=incompatible-pointer-types -Werror=designated-init -Wno-packed-not-aligned -fmacro-prefix-map=/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl=target-mipsel_24kc_musl -Werror -P -Umips -D__ASSEMBLY__ -DLINKER_SCRIPT -o arch/mips/kernel/vmlinux.lds arch/mips/kernel/vmlinux.lds.S
cc1: fatal error: symtab.h: No such file or directory
compilation terminated.
make[7]: *** [scripts/Makefile.build:373: arch/mips/kernel/vmlinux.lds] Error 1
make[6]: *** [scripts/Makefile.build:496: arch/mips/kernel] Error 2
make[5]: *** [Makefile:1801: arch/mips] Error 2
make[5]: Leaving directory '/home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16'
make[4]: *** [Makefile:24: /home/timo/Git/openwrt-r3g/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.10.16/.modules] Error 2
make[4]: Leaving directory '/home/timo/Git/openwrt-r3g/target/linux/ramips'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/home/timo/Git/openwrt-r3g/target/linux'
time: target/linux/compile#0.55#0.20#0.70
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:25: target/linux/compile] Error 1
make[2]: Leaving directory '/home/timo/Git/openwrt-r3g'
make[1]: *** [target/Makefile:18: /home/timo/Git/openwrt-r3g/staging_dir/target-mipsel_24kc_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/home/timo/Git/openwrt-r3g'
make: *** [/home/timo/Git/openwrt-r3g/include/toplevel.mk:230: world] Error 2```

This already was broken on the old 5.10 test as well, though.

File: package/network/services/wireguard/Makefile [hunk 1]
```diff
@@ -5,26 +5,43 @@
...
+# package/kernel/linux/modules/netsupport.mk and this package removed.
 
 include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=wireguard
 
-PKG_VERSION:=1.0.20201112
-PKG_RELEASE:=1
+PKG_CONFIG_DEPENDS:=CONFIG_LINUX_5_10
```
Comment (1715173329) [inline package/network/services/wireguard/Makefile:17]: This doesn't make sense.
Why not add it as a kernel module and rename the existing to wireguard-oot?

Comment (1715173329) [inline package/network/services/wireguard/Makefile:17]: This doesn't make sense.
Why not add it as a kernel module directly and rename the existing to wireguard-oot?

File: package/kernel/linux/modules/crypto.mk [hunk 2]
```diff
@@ -839,3 +841,121 @@ endef
 
 $(eval $(call KernelPackage,crypto-xts))
 
+define KernelPackage/crypto-curve25519
```
Comment (1715173329) [inline package/kernel/linux/modules/crypto.mk:844]: Please consider sequence.

File: package/network/services/wireguard/Makefile (hunk 1 above)
Comment (lipnitsk) [reply package/network/services/wireguard/Makefile:17]: there are dependencies in other modules on wireguard, including other repos (Luci), how would we manage that?

File: package/kernel/linux/modules/crypto.mk (hunk 2 above)
Comment (lipnitsk) [reply package/kernel/linux/modules/crypto.mk:844]: What do you mean by sequence?

Comment (1715173329) [reply package/kernel/linux/modules/crypto.mk:844]: e.g. `crypto-curve25519` should be put after `crypto-cts`.

File: package/network/services/wireguard/Makefile (hunk 1 above)
Comment (1715173329) [reply package/network/services/wireguard/Makefile:17]: Sorry for my poor Eng, I'll try to make it clear.

In `luci-proto-wireguard`, it depends on `kmod-wireguard` and `wireguard-tools`,
and we know, `wireguard-tools` actually depends on `kmod-wireguard` which is not added in it,
so just let `wireguard-tools` adjusts which package is needed, in-tree module or out-of-tree one,
like this: `+LINUX_5_4:kmod-wireguard-oot +!LINUX_5_4:kmod-wireguard`, then we can remove
`kmod-wireguard` in luci app.

Also, the oot one provided a meta package, see below:
https://github.com/openwrt/openwrt/blob/5bb9954826229782d4d285d65dfcca8a1c9e20d9/package/network/services/wireguard/Makefile#L50
https://github.com/openwrt/openwrt/blob/5bb9954826229782d4d285d65dfcca8a1c9e20d9/package/network/services/wireguard/Makefile#L53
We could reuse this as well, set `wireguard` as dependency in luci, and let it adjusts using the in-tree or oot,
but extra `VARIANT` is needed as build system will try to build the whole package by default.

Comment (lipnitsk) [reply package/network/services/wireguard/Makefile:17]: Okay I may do some work on this later today. @blocktrron I will let you know if a new commit is better for you to pick.

**Comment by lipnitsk** [comment]
@timocapa  try 5ea33837f863d6843b4cb0a8f173f1f6e0aeac2b

**Comment by timocapa** [comment]
> @timocapa  try 5ea33837f863d6843b4cb0a8f173f1f6e0aeac2b

Fixed it yeah

File: package/kernel/linux/modules/crypto.mk (hunk 2 above)
Comment (dengqf6) [reply package/kernel/linux/modules/crypto.mk:844]: > e.g. crypto-curve25519 should be put after crypto-cts.

So, "alphabetical ordering"?

Comment (lipnitsk) [reply package/kernel/linux/modules/crypto.mk:844]: Yeah I'll make the change. @dengqf6 do you want to coordinate mt7621 roll-out somehow - I can check your branch against my er-x. You can ping me on IRC if you want.

Comment (1715173329) [reply package/kernel/linux/modules/crypto.mk:844]: > So, "alphabetical ordering"?

Oh yes, sorry for my poor Eng.

File: package/network/services/wireguard/Makefile (hunk 1 above)
Comment (lipnitsk) [reply package/network/services/wireguard/Makefile:17]: https://github.com/openwrt/openwrt/pull/3885

File: package/kernel/linux/modules/crypto.mk (hunk 2 above)
Comment (lipnitsk) [reply package/kernel/linux/modules/crypto.mk:844]: https://github.com/openwrt/openwrt/pull/3885

**Pull request closed without merging**